src/utils.rs

```diff
@@ -96,6 +102,65 @@ impl PartialEq for CommandOutput {
     }
 }
 
+/// Allow the user to edit the given buffer in the system editor
+pub fn edit_temp(buffer: String) -> Result<String> {
+    let mut file = tempfile::NamedTempFile::new()?;
+    file.write_all(buffer.as_bytes())?;
+
+    let file = file.into_temp_path();
+    let fp = file.as_os_str().to_os_string();
+    edit(fp);
+    let buffer = read_file(&file)?;
+
+    file.close()?;
+    Ok(buffer)
+}
+
+/// Open the given path in the system editor
+pub fn edit(path: OsString) {
+    let editor_cmd = shell_words::split(&get_editor()).expect("Unmatched quotes found in $EDITOR.");
+
+    let command = create_command(&editor_cmd[0])
+        .expect("Unable to locate editor in $PATH.")
+        .stdin(Stdio::inherit())
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit())
+        .args(&editor_cmd[1..])
+        .arg(path)
+        .status();
```

Comment:
I'd prefer improving error handling on editing before going forward with this change (https://github.com/starship/starship/issues/3758 #3766)